### PR TITLE
Game update refactor

### PR DESCRIPTION
**Pull request by Endaris**
This is a bit of a conglomerate of changes (sorry for bad scoping...)
Effectively, `Game:update` looked quite convoluted with a lot of lower level code rather than function calls.
Additionally, shortcut processing was all over the place and some of the shortcuts were still not implemented.
I unified the shortcut handling into a function in a separate file that is just getting called.
I also extracted all lower level code besides the startup coroutine into functions.

Additionally I removed all references to `this_frame_messages` which was effectively a weird mechanism to bypass `serverQueue` for some things. As spectating functionality is broken right now anyway and you mentioned that you had issues with spectate, I figured it would be a good opportunity to unify the way we receive and process messages. Also one global less, yay.

### CODE REVIEW

File: inputManager.lua [hunk 1]
```diff
@@ -209,7 +209,7 @@ function inputManager:updateSystemKeys()
   -- systemKey
   self.isDown["SystemKey"] = self.isDown["Alt"] and self.isDown["Ctrl"] and self.isDown["Shift"]
   self.isUp["SystemKey"] = self.isUp["Alt"] and self.isUp["Ctrl"] and self.isUp["Shift"]
-  self.isPressed["SystemKey"] = math.min(self.isPressed["Alt"] or 0xFFFFFFFF, self.isPressed["Ctrl"] or 0xFFFFFFFF, self.isPressed["Shift"] or 0xFFFFFFFF)
+  self.isPressed["SystemKey"] = self.isPressed["Alt"] and self.isPressed["Ctrl"] and self.isPressed["Shift"]
```
[the remark this answers is not in the document]
Comment (Sankyr) [reply inputManager.lua:212]: Because is pressed stores the duration the button was held for, in order to combine these keys into a proper press that keeps track of press duration we need the min time all 3 were pressed. The max value is there so if the button wasn't pressed it will use some other value. 
@Endaris please keep that functionality (you can change the implementation if you want)

[the remark this answers is not in the document]
Comment (Endaris) [reply inputManager.lua:212]: Do we really care for how long they were pressed?
That seems like a superfluous feature.
Regardless, the previous implementation caused `if isPressed["SystemKey"] then` to be entered so long as *one* of the three modifiers was pressed, which was clearly not intended behaviour.

[the remark this answers is not in the document]
Comment (Sankyr) [reply inputManager.lua:212]: Ah, I don't know why I used that constant, it should have been -1 or some negative number, then it would behave as intended
But the point was for isPressed to store the duration of press.

File: Game.lua [hunk 2]
```diff
@@ -353,13 +303,15 @@ function Game:update(dt)
...
 
-  if self.server_queue and self.server_queue:size() > 0 then
-    logger.trace("Queue Size: " .. self.server_queue:size() .. " Data:" .. self.server_queue:to_short_string())
+  if self.backgroundImage then
+    self.backgroundImage:update(dt)
   end
-  this_frame_messages = {}
 
+  self:updateMouseVisibility(dt)
+  updateNetwork(dt)
```
[the remark this answers is not in the document]
Comment (Endaris) [reply Game.lua]: done

File: scenes/GameBase.lua [hunk 3]
```diff
@@ -229,20 +229,10 @@ function GameBase:runGameOver()
     do_messages() -- recieve messages so we know if the next game is in the queue
   end
 
-  local leftSelectMenu = false -- Whether a message has been sent that indicates a match has started or the room has closed
-  if this_frame_messages then
-    for _, msg in ipairs(this_frame_messages) do
```
[the remark this answers is not in the document]
Comment (Endaris) [reply scenes/GameBase.lua:234]: Not yet.
We have to do a complete pass through the network code anyway as at least half of that is either untested or confirmed not working so far.
The respective code is from the previous game over transition which is one of the things that should definitely be dying with the new UI anyway.
I'm almost sure even on beta there would be no significant adverse effect if this code was removed but I can't test it right now.

File: inputManager.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (Endaris) [reply inputManager.lua]: Fixed

File: scenes/GameBase.lua (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sankyr) [reply scenes/GameBase.lua:234]: this was only here because I ported it from mainloop, but ideally it would be isolated to the online game environment (or better yet, endaris will put it somewhere else)

File: network.lua [hunk 4]
```diff
@@ -140,14 +140,9 @@ function queue_message(type, data)
...
   elseif type == NetworkProtocol.serverMessageTypes.jsonMessage.prefix then
     local current_message = json.decode(data)
-    this_frame_messages[#this_frame_messages + 1] = current_message
     if not current_message then
       error(loc("nt_msg_err", (data or "nil")))
     end
-    if current_message.spectators then
-      spectators_string = spectator_list_string(current_message.spectators)
-      return
-    end
```
Comment (Sankyr) [inline network.lua:150]: why is this removed?

**Review by Sankyr — APPROVED** [review]

File: network.lua (hunk 4 above)
Comment (Endaris) [reply network.lua:150]: I don't want netcode to bypass `server_queue` for processing as it makes thing intransparent and confusing.
By removing it now, I won't forget about it later.
